go-controller/pkg/ovn/gateway/gateway.go

```diff
@@ -63,15 +63,23 @@ func GetGatewayPhysicalIPs(nbClient libovsdbclient.Client, gatewayRouter string)
 
 // CreateDummyGWMacBindings creates mac bindings (ipv4 and ipv6) for a fake next hops
 // used by host->service traffic
-func CreateDummyGWMacBindings(nbClient libovsdbclient.Client, gwRouterName string) error {
+func CreateDummyGWMacBindings(nbClient libovsdbclient.Client, gwRouterName string, netInfo util.NetInfo) error {
 	logicalPort := ovntypes.GWRouterToExtSwitchPrefix + gwRouterName
-	dummyNextHopIPs := node.DummyNextHopIPs()
-	smbs := make([]*nbdb.StaticMACBinding, len(dummyNextHopIPs))
-	for i := range dummyNextHopIPs {
+	ips := node.DummyNextHopIPs()
+	// In UDN, add static MAC bindings for host masquerade IPs.
+	// This is necessary because the masquerade network is directly
+	// attached to the external port of the gateway router,
+	// and neighbor discovery has to be avoided since these IPs
```

Comment:
hmm I'm a little surprised we didn't already do this, but I'm wondering how this is related to UDN or why it is a problem now?